packages/server-wallet/src/wallet/__test__/integration/join-channel.test.ts

```diff
@@ -185,3 +196,126 @@ describe('directly funded app', () => {
+    w.store.ledgers = {};
+    w.store.ledgerRequests = new LedgerRequests();
+
+    const someNonConflictingChannelNonce = 23364518;
+
+    // NOTE: Put a ledger Channel in the DB
+    ledger = await Channel.query(w.knex).insert(
+      channel({
+        signingAddress: bob().address,
+        channelNonce: someNonConflictingChannelNonce,
+        vars: [
+          stateWithHashSignedBy(
+            alice(),
+            bob()
+          )({
+            appDefinition: '0x0000000000000000000000000000000000000000',
+            channelNonce: someNonConflictingChannelNonce,
+            turnNum: 4,
+            outcome: simpleEthAllocation([{destination: bobP().destination, amount: BN.from(5)}]),
+          }),
+        ],
+      })
+    );
+
+    // Update the in-memory Ledgers table
+    w.__setLedger(ledger.channelId, ETH_ASSET_HOLDER_ADDRESS);
+
+    // Generate application channel
+    app = channel({
+      fundingStrategy: 'Ledger',
+    });
+
+    // Construct expected ledger update state
+    expectedUpdatedLedgerState = {
+      // eslint-disable-next-line
+      ...ledger.latest!,
+      turnNum: 6,
+      outcome: {
+        type: 'SimpleAllocation' as const,
+        // TODO: Avoid hardcoded response
+        assetHolderAddress: '0x0000000000000000000000000000000000000000',
+        allocationItems: [
+          {
+            destination: makeDestination(app.channelId), // Funds allocated to channel
+            amount: BN.from(5), // As per channel outcome
+          },
+        ],
+      },
+    };
+  });
+
+  const putTestChannelInsideWallet = async (args: Objection.PartialModelObject<Channel>) => {
+    const channel = await Channel.query(w.knex).insert(args);
+
+    await ObjectiveModel.insert(
+      {
+        type: 'OpenChannel',
+        participants: channel.participants,
+        data: {
+          targetChannelId: channel.channelId,
+          fundingStrategy: 'Ledger',
+        },
+        status: 'approved',
+      },
+      w.knex
+    );
+
+    return channel;
+  };
+
+  it('countersigns a prefund setup and automatically creates a ledger update', async () => {
+    const outcome = simpleEthAllocation([{destination: bobP().destination, amount: BN.from(5)}]);
+    const preFS0 = {turnNum: 0, outcome};
+    const preFS1 = {turnNum: 1, outcome};
+
+    const {channelId} = await putTestChannelInsideWallet({
+      ...app,
+      signingAddress: bob().address,
+      vars: [stateWithHashSignedBy(alice())(preFS0)],
+    });
+
+    const signedPreFS1 = stateWithHashSignedBy(bob())(preFS1);
+
+    await expect(w.joinChannel({channelId})).resolves.toMatchObject({
+      outbox: [
+        {
+          method: 'MessageQueued',
+          params: {
+            recipient: 'alice',
+            sender: 'bob',
+            data: {
+              signedStates: [
+                serializeState(stateWithHashSignedBy(bob())(signedPreFS1)),
+                serializeState(stateWithHashSignedBy(bob())(expectedUpdatedLedgerState)),
```

Comment:
Should this assert on the order?

It's a bit hard not to. We could define a custom `toHaveOutbox` matcher that handles this type of issue for you. I tried to create custom matchers in the past, but got stumped by typescript issues.